darc/darc_test.go

```diff
@@ -625,4 +630,128 @@ func TestParseIdentity(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, i.Proxy)
 	require.Equal(t, in, i.String())
+
+	in = "did:method:WRfXPg8dantKVubE3HX8pw"
+	i, err = ParseIdentity(in)
+	require.Error(t, err)
+
+	in = "did:sov:WRfXPg8dantKVubE3HX8pw"
+	i, err = ParseIdentity(in)
+	require.NoError(t, err)
+	require.NotNil(t, i.DID)
+	require.Equal(t, strings.Split(in, ":")[2], i.DID.DID)
+	require.Equal(t, "sov", i.DID.Method)
+}
+
+func TestDarc_EvolveToDIDDarc(t *testing.T) {
+	dataPublisher := createDarc(1, "Patient Data Publisher Darc")
+
+	patientIdentity := NewIdentityDID("WRfXPg8dantKVubE3HX8pw", "sov")
+	patientRules := InitRules([]Identity{patientIdentity}, []Identity{patientIdentity})
+	patientDarc := NewDarc(patientRules, []byte("Patient Darc"))
+	err := patientDarc.EvolveFrom(dataPublisher.darc)
+	require.NoError(t, err)
+
+	r, patientDarcBuf, err := patientDarc.MakeEvolveRequest(dataPublisher.owners[0])
+	require.NoError(t, err)
+
+	err = r.Verify(dataPublisher.darc)
+	require.NoError(t, err)
+	patientDarcEvolved, err := r.MsgToDarc(patientDarcBuf)
+	require.NoError(t, err)
+	require.Equal(t, patientDarc.GetID(), patientDarcEvolved.GetID())
+
+	err = patientDarcEvolved.VerifyWithCB(func(s string, latest bool) *Darc {
+		if s == dataPublisher.darc.GetIdentityString() {
+			return dataPublisher.darc
+		}
+		return nil
+	}, true)
+	require.NoError(t, err)
+}
+
+func TestDarc_VerifyWithDIDDarc(t *testing.T) {
+	patientIdentity := NewIdentityDID("CvjH5UcquEEqsz5LrZcfvM", "sov")
+
+	patientRules := InitRules([]Identity{patientIdentity}, []Identity{patientIdentity})
+	patientRules.AddRule("Read", expression.InitOrExpr(patientIdentity.String()))
+
+	patientDarc := NewDarc(patientRules, []byte("Patient Darc"))
+
+	signature := "4b61e1e81fb17145d9a560724f16902673a5b9c86cb5bd330d51722fcd66b9311b1821e319e9160062f88eab476a9e3bdc5e770f85b194eec91585fea24d400e"
+	sigBuf := make([]byte, hex.DecodedLen(len([]byte(signature))))
+	hex.Decode(sigBuf, []byte(signature))
+
+	// overriding the global in the test
+	didResolver = &IndyCLIDIDResolver{
+		Path:            "./mock_resolver_outputs/test_resolver",
```

Comment:
Instead of running a test_resolver from the unit test, teach IndyCLIDIDResolver that if it has an io.Reader set, then it should read from that instead of spawning `indy-cli`. Then you can have the canned results right here in the unit test.